emmet-builders/emmet/builders/qchem/molecules.py

```diff
@@ -224,29 +223,38 @@ def process_item(self, items: List[Dict]) -> List[Dict]:
         """
 
         tasks = [TaskDocument(**task) for task in items if task["is_valid"]]
+        if len(tasks) == 0:
+            return list()
         formula = tasks[0].formula_alphabetical
         task_ids = [task.task_id for task in tasks]
         self.logger.debug(f"Processing {formula} : {task_ids}")
         molecules = list()
 
         for group in self.filter_and_group_tasks(tasks):
             try:
-                molecules.append(MoleculeDoc.from_tasks(group))
+                doc = MoleculeDoc.from_tasks(group)
+                molecule_id = "{}-{}-{}".format(
+                    doc.coord_hash,
+                    str(int(doc.charge)).replace("-", "m"),
+                    doc.spin_multiplicity,
+                )
+                doc.molecule_id = molecule_id
+                molecules.append(doc)
```

Comment:
@espottesmith when testing out the build pipeline on solvated ion clusters I found a bug in this section of code. For monatomic ions with the same charge, this will result in duplicate `molecule_id` b/c the coordinates, charge, and multiplicity can all be identical.

I worked around the issue by doing this; although there is probably a more robust / general solution:
```
        for group in self.filter_and_group_tasks(tasks):
            try:
                # TODO - this molecule_id formula will return identical ids for single atoms
                # that have the same charge and spin
                doc = MoleculeDoc.from_tasks(group)
                # use something different for single atoms
                if len(doc.composition)==1:
                    molecule_id = MPculeID("{}-{}-{}".format(
                        doc.species_hash,
                        str(int(doc.charge)).replace("-", "m"),
                        doc.spin_multiplicity,
                    ))
                else:
                    molecule_id = MPculeID("{}-{}-{}".format(
                        doc.coord_hash,
                        str(int(doc.charge)).replace("-", "m"),
                        doc.spin_multiplicity,
                    ))
                doc.molecule_id = molecule_id
                molecules.append(doc)
```

I tried just changing the `coord_hash` to `species_hash` for all tasks, but that can result in duplicated IDs for other reasons, so I think it's important to keep the coordinates in there in most cases.

Reply:
Yeah, I realized this a while back and hadn't had a chance to fix it. The solution that I'm going to go for is actually just making MPculeID a little longer and including the formula. I actually also think that'll help it be more readable. The hash still won't be easily interpreted, but the user will at least have a clue what they're looking at if the formula is there.

Reply:
Sounds good. If you have any prototype code for that (even if very rough), feel free to send my way and I can test it on my builds (but if you haven't got there yet, no worries)

Reply:
I just realized we also need to consider what happens if we have the same single atom / ion in different solvents. My solutions breaks in that case - all solvents collapse to the same MP-ID which results in calcs getting lost in the build pipeline.

Reply:
That should be fine, I think? Since I've changed how level of theory is handled, and molecules in different solvents should be collapsed into a single document anyways. You're right that this will be treated somewhat differently in the association stage, because in general molecules in different solvents won't have identical structures. But I don't believe it's inconsistent. Will test this to confirm.

Reply:
Yeah as long as molecules in different solvents wind up in the same document / ID this should be OK. I'm not seeing that happen in my current build pipeline testing (it seems that only Water gets past the association stage), but I haven't had time to troubleshoot much yet

Reply:
If you can send me a minimal test case (let's say one molecule in 2 or 3 different solvents), that'd actually be a big help.